app/code/core/Mage/Core/Helper/Abstract.php

```diff
@@ -343,6 +343,7 @@ public function jsQuoteEscape($data, $quote = '\'')
      */
     public function quoteEscape($data, $addSlashes = false)
     {
+        $data = (string)$data;
```

Comment:
Is this more efficient?
```suggestion
        if (!$data) {
            return $data;
        }
```

Reply:
Yes, but this is due to a third party code not up to date, so I close this one?

Reply:
I think we still need to fix it.

Reply:
I think we should close this one. the function is clearly marked to accept a string and there seems there are no misuse of the type inside our code